### DISCUSSION TITLE
Not able to set it up

### DISCUSSION THREAD

**Discussion by akhil-325** [pranjal-joshi/Screeni-py#158] [Q&A]
```
akhil@Akhils-MacBook-Air ~ % /Users/akhil/Downloads/screenipy.run ; exit;
zsh: illegal hardware instruction  /Users/akhil/Downloads/screenipy.run

Saving session...
...copying shared history...
...saving history...truncating history files...
...completed.

[Process completed]
```

Can someone guide me. I have given it the execution access. When i double click the screenipy.run file it shows the above code.
Not able to figure out the issue. Mac os monterey 12.1 m1

**Comment by pranjal-joshi** [comment]
Hi @akhil-325 

Are you running on Apple silicon (M1/M2) chip?

**Reply by akhil-325** [reply]
yes m1
is that an issue?

also, it was working earlier, about 2-3mnths ago
but now isnt, so downloaded the new versions, still not working

**Reply by pranjal-joshi** [reply]
Not an issue.
The `.run` file is generated on cloud which works with older Intel CPUs, thus makes the same file incompatible with M1.
I myself use Mac M1 and prefer running it through python script.
If you are interested in experimenting, please try searching about `rossetta` platform and how it can run intel based files on the M1 platform.
Let me know if you find a solution for this.

**Reply by akhil-325** [reply]
Oh okay. Will search.

new to github, can i also run it through python? not sure if i have access to the code

**Reply by akhil-325** [reply]
also why was i able to run it couple of months ago? i had used it on the same m1 mac only

**Reply by pranjal-joshi** [reply]
> Oh okay. Will search.
> 
> new to github, can i also run it through python? not sure if i have access to the code

Yes. This GitHub repo contains all the code you need.
You can quickly see any YouTube video for "Cloning a GitHub repo"
Once cloned, follow point number 2 & 3 from https://github.com/pranjal-joshi/Screeni-py/blob/main/CONTRIBUTING.md

**Reply by pranjal-joshi** [reply]
> also why was i able to run it couple of months ago? i had used it on the same m1 mac only
> 

That's strange!

Recently added AI based functionality using TensorFlow framework.
Seems like building a run file after adding that feature is causing this problem.

**Reply by akhil-325** [reply]
Thanks for the help.

**Reply by pranjal-joshi** [reply]
Can you please post output of this command to solve this issue?

`lipo -info screenipy.run`

**Reply by akhil-325** [reply]
`fatal error: /Library/Developer/CommandLineTools/usr/bin/lipo: can't open input file: screenipy.run (No such file or directory)`

Edit: I guess this didnt reach the correct file, retried by going to the folder

`Non-fat file: screenipy.run is architecture: x86_64`

**Reply by parthweprom** [reply]
I am also facing same issue. Is there any update on it.

**Reply by pranjal-joshi** [reply]
@parthweprom 

Install python environment and setup dependencies
The executable with M1/M2 may have some issues.
